test/modules/layers/scenegraph-layer.spec.js

```diff
@@ -22,23 +22,102 @@ import test from 'tape-catch';
 import {testLayer, generateLayerTests} from '@deck.gl/test-utils';
 
 import {ScenegraphLayer} from '@deck.gl/mesh-layers';
+import {GroupNode, ModelNode, CubeGeometry} from '@luma.gl/core';
 
 import * as FIXTURES from 'deck.gl-test/data';
 
+const fs = `\
+varying vec4 vColor;
+
+void main(void) {
+  gl_FragColor = vColor;
+}
+`;
+
+const vs = `\
+uniform float sizeScale;
+
+attribute vec3 positions;
+
+attribute vec3 instancePositions;
+attribute vec2 instancePositions64xy;
+attribute vec4 instanceColors;
+attribute vec3 instanceTranslation;
+
+varying vec4 vColor;
+
+void main(void) {
+  vColor = instanceColors / 255.0;
+
+  vec3 pos = (positions + instanceTranslation) * sizeScale;
+  pos = project_size(pos);
+
+  vec4 position_commonspace;
+  gl_Position = project_position_to_clipspace(instancePositions, instancePositions64xy, pos, position_commonspace);
+}
+`;
+
+class MockGLTFAnimator {
+  constructor() {
+    this.animation0 = {};
+    this.animation1 = {};
+    this.animation2 = {name: 'name'};
+  }
+
+  getAnimations() {
+    return [this.animation0, this.animation1, this.animation2];
+  }
+
+  animate() {}
+}
+
 test('ScenegraphLayer#tests', t => {
   const testCases = generateLayerTests({
     Layer: ScenegraphLayer,
     sampleProps: {
       data: FIXTURES.points,
       getPosition: d => d.COORDINATES,
-      scenegraph: ''
+      getTranslation: d => [0, 0, 2],
+      pickable: false,
+      sizeScale: 50,
+      scenegraph: true,
+      _animations: {
+        '*': {
+          playing: true,
+          speed: 10
+        },
+        1: {
+          playing: true,
+          speed: 20
+        },
+        name: {
+          playing: true,
+          speed: 30
+        }
+      },
+      getScene: (_scenegraph, {gl}) => {
+        return new GroupNode([
+          new ModelNode(gl, {
+            geometry: new CubeGeometry(),
+            vs,
+            fs,
+            modules: ['project32'],
+            isInstanced: true
+          })
+        ]);
+      },
+      getAnimator: () => new MockGLTFAnimator()
     },
     assert: t.ok,
     onBeforeUpdate: ({testCase}) => t.comment(testCase.title),
-    onAfterUpdate: ({layer, subLayers}) => {
-      // if (layer.props.mesh) {
-      //   t.ok(layer.getModels().length > 0, 'Layer should have models');
-      // }
+    onAfterUpdate: ({layer}) => {
+      if (Object.keys(layer.props).length > 0) {
```

Comment:
check `layer.props.scenegraph` and `layer.props._animation`

Reply:
Done.